### DISCUSSION TITLE
Converting circular structure to JSON when updating a property

### DISCUSSION THREAD

**Discussion by Pulstarter** [livewire/livewire#3003] [7. Bugs (archived)]
### Description

When a blade component with alpine has a dynamic property and this property changes, it throws the following error:

Uncaught TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'Object'
    --- property 'defer' closes the circle
    at JSON.stringify (<anonymous>)
...

### Exact steps to reproduce

Create a blade component with a dynamic prop and assing it to a x-data property:

if the component updates, the error appear.

### Stripped-down, copy-pastable code snippets

@props([
'id' => Str::random(),
'options' => [],
])

<div x-data="{
{value: @entangle($attributes->wire('model')),
                  options: [],
                  myid: '{{$id}}'
}">

</div>

**Context**
- Livewire version: [2.4.4]
- Laravel version: [8.42.1]
- Alpine version: [2.8.2]
- Browser: [Chrome]

**Comment by joshhanley** [comment]
@Pulstarter can you please provide all the code (that we can copy paste) or a repo that demonstrates the issue so it can be investigated? 

Thanks!

**Comment by Pulstarter** [comment]
Hi @joshhanley !

I have found a workaround without using @entangle and wire:model, however I would like to know what I am doing wrong.

I have uploaded both versions of the blade component that I am using.  It's an implementation of the select2 plugin:

[https://github.com/Pulstarter/livewire-select2](https://github.com/Pulstarter/livewire-select2)

To use the ko file:

`<x-select-ko
                        wire:model="form.tipology_id"
                        name="form.tipology_id"
                        label="{{__('Tipology')}}"
                        :options="$_tipologies"
                        required
                        inline
                    />`

To use the workaround:

`<x-select
                        :value="$form['tipology_id']"
                        name="form.tipology_id"
                        label="{{__('Tipology')}}"
                        :options="$_tipologies"
                        required
                        inline
                    />`

Both works fine on the initial load, but It you update the $_tipologies array on the livewire component, then the error appears.
